### PR TITLE
Implemented the pheromone-reduction (Issue 77)

### PR DESCRIPTION
**Pull request by tis22**
We decided that we want to reduce the pheromones by a selected factor (`reducing_factor`) after each epoch.
By this pheromones which are 0 will stay 0, pheromones != 0 will be reduced weighted by their amount over time (higher amount of pheromones = higher reduction).
There is an argument to adjust a limit from which the value is considered 0 also (`zero_threshold`).

The tests in the test-method prove if the reduction is performed correctly for the "from colony" pheromones (first layer, negative values) and the "from food" pheromones (second layer, positive values).
Moreover the tests prove that pheromones are set to zero if they are greater/less than the threshold.

### CODE REVIEW

**Comment by tis22** [comment]
Removed the timestamps-array, tests need to be corrected.

File: test/test_pheromone.py [hunk 1]
```diff
@@ -29,4 +28,36 @@ def test_get_pheromone_level():
...
+    assert levels == {'coming from colony': - 0.5, 'coming from food': 0.5}
+
+    # Second reduction
+    pheromone.reduce_pheromones(reducing_factor = 0.5)
+    levels = pheromone.get_pheromone_level(pos)
+    assert levels == {'coming from colony': - 0.25, 'coming from food': 0.25}
+
+    # Further reductions
+    for _ in range(5):
+        pheromone.reduce_pheromones(reducing_factor = 0.5)
```
Comment (AlhaririAnas) [inline test/test_pheromone.py:59]: instead of for loop: 
```suggestion
        pheromone.reduce_pheromones(reducing_factor = 0.5**5)
```

**Review by AlhaririAnas — CHANGES_REQUESTED** [review]

File: test/test_pheromone.py (hunk 1 above)
Comment (tis22) [reply test/test_pheromone.py:59]: Thanks for this improvement, I didn't think of this solution at the time, but I knew there was something better than the for-loop!

**Comment by tis22** [comment]
The for-loop need to be removed when we use 0.5**5.

**Review by AlhaririAnas — APPROVED** [review]
I have reviewed all the changes, and everything looks good. Nice job!

**Pull request merged**